code/modules/projectiles/projectile/beams.dm

```diff
@@ -64,6 +68,17 @@
 /obj/item/projectile/beam/pulse/shot
 	damage = 40
 
+/obj/item/projectile/beam/pulse/heavy
+	name = "heavy pulse laser"
+	icon_state = "pulse1_bl"
+	var/life = 20
+
+/obj/item/projectile/beam/pulse/heavy/on_hit(atom/target, blocked = 0, hit_zone)
+	life -= 10
+	if(life > 0)
+		. = -1
+	..()

```

Comment:
This replaces /pulse/heavy/Bump() which was causing the projectile to hit targets twice. The projectile's behavior is unchanged.